io/include/pcl/io/file_io.h

```diff
@@ -234,7 +234,7 @@ namespace pcl
   template <typename Type> inline
   std::enable_if_t<std::is_floating_point<Type>::value>
   copyValueString (const pcl::PCLPointCloud2 &cloud,
-                   const unsigned int point_index, 
+                   const pcl::index_t point_index, 
```

Comment:
I assume implicit conversion will happen here with no issues API wise. There will be ABI breakage. Since there's no ABI guarantees for between minor versions I would say let's go with it and not care about deprecation.

Reply:
Replacing unsigned int with `uindex_t` will make sure no ABI issues, but it's technically index_t. I'll mark it as ABI break here.

Reply:
Why was `uindex_t` actually introduced? Indices should always(?) be nonnegative anyway. When should `index_t` be used and when `uindex_t`? I did not find any information on this in the documentation.

Reply:
it's an alias to an unsigned representation of whatever index_t is, even if index_t is unsigned. It's a helper type for assisting in the widespread migration to index_t. 

Always use index_t. Use uindex_t only you need a guaranteed unsigned type with the same bit length of index_t.

Reply:
> Indices should always(?) be nonnegative anyway

Yes, but 2 counterpoints:
1. the behaviour till 1.11 was to use `int` much more than `uint` for indices
2. openmp pragma needs the index to be signed, which means this should be an opt-in for the extreme case scenario where casting to signed will overflow

Maybe a PR with these 2 reasons would be nice :laughing: Sorry bit busy reviewing loads of code now and was tardy with my code before.